cloud/hack/test-integration.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The KubeEdge Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT=$(unset CDPATH && cd $(dirname "${BASH_SOURCE[0]}")/../../ && pwd)
+export KUBE_CONFIG=""
+export KUBE_APISERVER_URL="http://localhost:8080"
+export TESTNS="crd-test"
+
+runTests() {
+  logStatus "Applying device crd yamls"
+  kubectl create -f "${SCRIPT_ROOT}/build/crds/devices/devices_v1alpha1_devicemodel.yaml"
+  kubectl create -f "${SCRIPT_ROOT}/build/crds/devices/devices_v1alpha1_device.yaml"
```

Comment:
Could you pls isolate those resource in the namespace as well?

```bash
 make test-integration
hack/test-integration.sh
+++ [0716 10:04:16] Applying device crd yamls
Error from server (AlreadyExists): error when creating "/go/src/github.com/kubeedge/kubeedge/build/crds/devices/devices_v1alpha1_devicemodel.yaml": customresourcedefinitions.apiextensions.k8s.io "devicemodels.devices.kubeedge.io" already exists
Makefile:40: recipe for target 'test-integration' failed
make: *** [test-integration] Error 1
```

Reply:
the crd definitions are cluster wide resourcesand not tied to a particular namespace.

Reply:
How about some pre-check before the creation?

Reply:
added the check. thanks.